modules/assim.sequential/R/sda.enkf_MultiSite.R

```diff
@@ -149,7 +149,14 @@ sda.enkf.multisite <- function(settings,
   obs.mean <- obs.mean[sapply(lubridate::year(names(obs.mean)), function(obs.year) obs.year %in% (assim.sda))] #checks obs.mean dates against assimyear dates
   obs.cov <- obs.cov[sapply(lubridate::year(names(obs.cov)), function(obs.year) obs.year %in% (assim.sda))] #checks obs.cov dates against assimyear dates
   #checking that there are dates in obs.mean and adding midnight as the time
-  obs.times <- names(obs.mean)
+  if(settings$state.data.assimilation$forecast.time.step == "year"){
+    obs.times <- PEcAnAssimSequential::obs_timestep2timepoint(settings$state.data.assimilation$Obs_Prep$start.date,
```

Comment:
Conceptually, I wonder if it would make more sense to do this fix in the assembler (i.e. add more "empty" dates to obs.mean)? To me, it seems cleaner, more flexible, and more clear to users to have the SDA itself always run exactly what days its told to, rather than having to document the behaviour that the SDA function adds in dates magically. The automatic adding in of dates could potentially mess up someone who is trying to run "non-standard" assimilation experiments.

Reply:
Fixed.